11_kaprekars_number.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Implementation for RubyMonk Primer Problem 11 Kaprekar's Number
+# Problem Statement:
+# 9 is a Kaprekar number since
+# 9 ^ 2 = 81 and 8 + 1 = 9
+# 297 is also Kaprekar number since
+# 297 ^ 2 = 88209 and 88 + 209 = 297.
+# In short, for a Kaprekar number k with n-digits, if you square it and add
+# the right n digits to the left n or n-1 digits, the resultant sum is k.
+# Find if a given number is a Kaprekar number
+class KaprekarsNumber
+  # rubocop:disable Naming/MethodParameterName
+  def kaprekar?(k)
```

Comment:
I think this is another place that might benefit from passing `k` at instantiation, but I'm having trouble comprehending the logic well enough to say how much of a win it is.  (Not just your logic; their explanation isn't very helpful to me either.)

Reply:
Again another case where RubyMonk just provided a global method to implement and I added the wrapping class for better test style.